src/url.js

```diff
@@ -92,12 +92,16 @@ export function getWinOrigin(win) {
  * testing by freezing the object.
  * @param {string} url
  * @param {boolean=} opt_nocache
+ *   Cache is altogether ignored on ESM builds, since an <a> is not used.
  * @return {!Location}
  */
 export function parseUrlDeprecated(url, opt_nocache) {
   if (!a) {
     a = /** @type {!HTMLAnchorElement} */ (self.document.createElement('a'));
-    cache = self.__AMP_URL_CACHE || (self.__AMP_URL_CACHE = new LruCache(100));
+    a.href = '';
```

Comment:
The old proposed code is actually better here. We don't need to worry about `a` having a different base HREF here, since both `a` and `URL` originate from the same window context (this current one).